common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java

```diff
@@ -362,17 +412,78 @@ public Bundle getRequestBundleForHello(@NonNull final CommandParameters paramete
         return requestBundle;
     }
 
-    public Bundle getRequestBundleForAcquireTokenSilent(final SilentTokenCommandParameters parameters) {
+    public Bundle getRequestBundleForAcquireTokenInteractive(@NonNull final InteractiveTokenCommandParameters parameters,
+                                                             @Nullable final String negotiatedBrokerProtocolVersion) {
+        final BrokerRequest brokerRequest = brokerRequestFromAcquireTokenParameters(parameters);
+        final Bundle requestBundle = new Bundle();
+
+        if (BrokerProtocolVersionUtil.canCompressBrokerPayloads(negotiatedBrokerProtocolVersion)) {
+            try {
+                final String jsonRequestString = sRequestAdapterGsonInstance.toJson(brokerRequest, BrokerRequest.class);
+                byte[] compressedBytes = compressString(jsonRequestString);
+                Logger.info(TAG, "Broker Request, raw payload size:"
+                        + jsonRequestString.getBytes().length + " ,compressed bytes size: " + compressedBytes.length
+                );
+                requestBundle.putByteArray(
+                        AuthenticationConstants.Broker.BROKER_REQUEST_V2_COMPRESSED,
+                        compressedBytes
+                );
+
+            } catch (IOException e) {
+                Logger.error(TAG, "Compression to bytes failed, sending broker request as String", e);
+                requestBundle.putString(
+                        AuthenticationConstants.Broker.BROKER_REQUEST_V2,
+                        sRequestAdapterGsonInstance.toJson(brokerRequest, BrokerRequest.class)
+                );
+            }
+        } else {
+            Logger.info(TAG, "Broker protocol version: " + negotiatedBrokerProtocolVersion +
+                    " lower than compression changes, sending as string"
+            );
+            requestBundle.putString(
+                    AuthenticationConstants.Broker.BROKER_REQUEST_V2,
+                    sRequestAdapterGsonInstance.toJson(brokerRequest, BrokerRequest.class)
+            );
+        }
+        return requestBundle;
```

Comment:
I see almost same code below in `getRequestBundleForAcquireTokenSilent`. I think we can move this logic to its own method.

Reply:
thanks for spotting, addressed via https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/891/commits/c8ba307d2e21dec24a677bdc2a04db7ef91608d5